shared/profile/prove-enter-username.native.js

```diff
@@ -1,11 +1,142 @@
 // @flow
-import {Component} from 'react'
+import React, {Component} from 'react'
+import {platformText} from './prove-enter-username.shared'
+import {Box, Icon, Text, Button, Input, PlatformIcon} from '../common-adapters'
+import {constants} from '../constants/types/keybase-v1'
+import {globalStyles, globalColors, globalMargins} from '../styles/style-guide'
+import openURL from '../util/open-url'
+
+import type {PlatformsExpandedType} from '../constants/types/more'
 import type {Props} from './prove-enter-username'
 
-class Render extends Component<void, Props, void> {
+type State = {
+  username: string
+}
+
+function UsernameTips ({platform}: {platform: PlatformsExpandedType}) {
+  if (platform === 'hackernews') {
+    return (
+      <Box style={styleInfoBanner}>
+        <Text backgroundMode='Information' type='BodySmallSemibold'>&bull; You must have karma &ge; 2</Text>
+        <Text backgroundMode='Information' type='BodySmallSemibold'>&bull; You must enter your uSeRName with exact case</Text>
+      </Box>
+    )
+  }
+
+  return null
+}
+
+function customError (error: string, code: ?number) {
+  if (code === constants.StatusCode.scprofilenotpublic) {
+    return <Box style={{...globalStyles.flexBoxColumn, justifyContent: 'center', alignItems: 'center'}}>
+      <Text style={{...styleErrorBannerText, marginLeft: globalMargins.small, marginRight: globalMargins.small}} type='BodySmallSemibold'>You haven't set a public "Coinbase URL". You need to do that now.</Text>
+      <Box style={{...globalStyles.flexBoxRow, alignItems: 'center'}} onClick={() => openURL('https://www.coinbase.com/settings#payment_page')}>
+        <Text style={styleErrorBannerText} type='BodySmallSemibold'>Go to Coinbase</Text>
+        <Icon type='iconfont-open-browser' style={{color: globalColors.white_40, marginLeft: 4}} />
+      </Box>
+    </Box>
+  }
+  return <Text style={styleErrorBannerText} type='BodySmallSemibold'>{error}</Text>
+}
+
+class Render extends Component<void, Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      username: '',
+    }
+  }
+
+  handleUsernameChange (username: string) {
+    if (this.props.onUsernameChange) {
+      this.props.onUsernameChange(username)
+    }
+    this.setState({username})
+  }
+
+  handleContinue () {
+    this.props.onContinue(this.state.username)
+  }
+
   render () {
-    return null
+    const {floatingLabelText, hintText} = platformText[this.props.platform]
+    return (
+      <Box style={styleContainer}>
+        <Box style={styleCancel}>
+          <Text type='BodyPrimaryLink' style={{position: 'absolute', top: 0, color: globalColors.blue}} onClick={this.props.onCancel}>Cancel</Text>
+        </Box>
+        {this.props.errorText && <Box style={styleErrorBanner}>{customError(this.props.errorText, this.props.errorCode)}</Box>}
```

Comment:
do `!!this.props.errorText` so you don't get cryptic errors like '' is not wrapped in a <Text>